### PR TITLE
patroni: init at 1.6.4

### PR DESCRIPTION
**Pull request by invokesus**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sql/patroni/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    pyyaml
+    tzlocal
+    urllib3
+    ydiff
+  ];
+
+  preCheck = "export HOME=$(mktemp -d)";
+
+  meta = with lib; {
+    homepage = https://patroni.readthedocs.io/en/latest/;
```
Comment (doronbehar) [inline pkgs/servers/sql/patroni/default.nix]:
```suggestion
    homepage = "https://patroni.readthedocs.io/en/latest/";
```

File: pkgs/servers/sql/patroni/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    psycopg2
+    python-dateutil
+    python-etcd
+    pyyaml
+    tzlocal
+    urllib3
+    ydiff
+  ];
+
+  preCheck = "export HOME=$(mktemp -d)";
```
Comment (doronbehar) [inline pkgs/servers/sql/patroni/default.nix]:
```suggestion
  # Fix tests by preventing them to write to /home-shlter
  preCheck = "export HOME=$(mktemp -d)";
```

File: pkgs/servers/sql/patroni/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  name = "patroni-${version}";
+  version = "1.6.4";
+
+  src = fetchurl {
+    url = "https://github.com/zalando/patroni/archive/v${version}.tar.gz";
+    sha256 = "0a8pzy3sf164byz4wkjcgm6yi45y2rkaf8jvzlhnbq3rdq68pb0x";
+  };
+
+  postPatch = ''
+    # cdiff renamed to ydiff; remove when patroni source reflects this.
```
Comment (doronbehar) [inline pkgs/servers/sql/patroni/default.nix]: Please put the comments outside the hook.

File: pkgs/servers/sql/patroni/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{ lib, pythonPackages, fetchurl }:
+
+pythonPackages.buildPythonApplication rec {
+  name = "patroni-${version}";
+  version = "1.6.4";
+
+  src = fetchurl {
```
Comment (doronbehar) [inline pkgs/servers/sql/patroni/default.nix]: Use `fetchFromGithub`.

File: pkgs/servers/sql/patroni/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{ lib, pythonPackages, fetchurl }:
+
+pythonPackages.buildPythonApplication rec {
+  name = "patroni-${version}";
```
Comment (doronbehar) [inline pkgs/servers/sql/patroni/default.nix]: The `name` attr will be applied automatically.

```suggestion
  pname = "patroni";
```

File: pkgs/servers/sql/patroni/default.nix [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  postPatch = ''
+    # cdiff renamed to ydiff; remove when patroni source reflects this.
+    for i in requirements.txt patroni/ctl.py tests/test_ctl.py; do
+      substituteInPlace $i --replace cdiff ydiff
+    done
+  '';
+
+  nativeBuildInputs = with pythonPackages; [
```
Comment (doronbehar) [inline pkgs/servers/sql/patroni/default.nix]: Not a big deal, but I'd use `checkInputs` instead:

```suggestion
  checkInputs = with pythonPackages; [
```

**Comment by invokesus** [comment]
Thanks for reviewing. I amended and force-pushed. :)

**Review by doronbehar — APPROVED** [review]

**Pull request merged**